### DISCUSSION TITLE
useVirual become unresponsive

### DISCUSSION THREAD

**Discussion by spacepolice10** [TanStack/virtual#402] [Q&A]
Hello,

Is there a way to find out why my application becomes unresponsive if I use useVirtual instead of useWindowVirtual? I detected that every element rerenders thousands times every second. But my component structure is no different from the one that used as an example on a Tanstack Virtual website and this example has no such troubles. Is there a way to find exact element causing such a problem? Maybe someone stumbled upon something like that?

**Comment by piecyk** [comment]
Hi, without minimal, reproducible example it's hard to tell what's is causing this.

**Reply by spacepolice10** [reply]
No matter what kind of data used it happens. I used the exact setup from here https://tanstack.com/virtual/v3/docs/examples/react/infinite-scroll 

Probably, the problem is not setting determined height (i changed that parameter). **If I use 100vh, auto or 100% on parrentRef container and useEffect hook to load new pages the application becomes unresponsive.** That's strange. I'd like to utilize useVirtualizer instead of useWindowVirtualizer because it restore scrolling position correctly but if I understand right, you can't stretch full screen.

Thank you for response

**Reply by piecyk** [reply]
Yes, in that case you are rendering every element from list.

**Reply by spacepolice10** [reply]
ah, I see. probably that's why it restores scroll position exactly as it supposed to be )
